### PR TITLE
Start implementing the data model

### PR DESCRIPTION
**Pull request by grtlr**
⚠️ Superseded by #55.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "chronicle"
-version = "0.1.0"
```
Comment (Alex6323) [inline Cargo.toml:3]: why that?

Comment (grtlr) [reply Cargo.toml:3]: I want to pre-release some Docker images to the GitHub container registry and I want to make sure that no one actually tries to use them 😁

Comment (Alexandcoats) [reply Cargo.toml:3]: Does this really determine the docker version?

File: Cargo.toml [hunk 2]
```diff
@@ -19,13 +19,15 @@ name = "inx-chronicle"
 path = "bin/inx-chronicle/src/main.rs"
 
 [dependencies]
+bee-message-stardust = { package = "bee-message", version = "0.2", git = "https://github.com/iotaledger/bee", branch = "shimmer-develop", default-features = false, features = ["std", "serde1", "dto"], optional = true }
```
Comment (Alexandcoats) [inline Cargo.toml:22]: Separate optional dependencies?

File: Cargo.toml [hunk 3]
```diff
@@ -19,13 +19,15 @@ name = "inx-chronicle"
...
+
 async-recursion = { version = "1.0", default-features = false }
 async-trait = { version = "0.1", default-features = false }
-clap = { version = "3.1", default-features = false, features = [ "derive", "std" ] }
+clap = { version = "3.1", default-features = false, features = ["derive", "std"] }
 dotenv = { version = "0.15", default-features = false }
 env_logger = { version = "0.9", default-features = false, features = ["termcolor", "atty", "humantime"] }
 futures = { version = "0.3", default-features = false }
-inx = { git = "https://github.com/iotaledger/inx", version = "0.2", default-features = false }
+inx = { git = "https://github.com/grtlr/inx", branch = "types", version = "0.2.0", default-features = false, features = ["types"] }
```
Comment (Alexandcoats) [inline Cargo.toml:30]: Use a patch instead?

File: src/db/model/mod.rs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#[cfg(feature = "stardust")]
+/// Models for Stardust layouts.
+pub mod stardust;
+
+use serde::Serialize;
+
+// TODO: Add `serde::Deserialize` constraint.
```
Comment (Alexandcoats) [inline src/db/model/mod.rs:10]: Probably just use `DeserializeOwned`.

File: src/db/model/stardust.rs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+use bee_message_stardust as stardust;
+use serde::{Deserialize, Serialize};
+
+use super::Model;
+
+/// Model for the [`Message`].
+#[derive(Serialize, Deserialize)]
+pub struct Message {
```
Comment (Alexandcoats) [inline src/db/model/stardust.rs:11]: We should probably go ahead and include optional metadata.

File: src/db/model/stardust.rs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    const COLLECTION: &'static str = "stardust_messages";
+}
+
+/// Model
+#[derive(Serialize, Deserialize)]
+pub struct Milestone {
+    /// The milestone index.
+    pub milestone_index: u32,
+    /// The timestamp of the milestone.
+    pub milestone_timestamp: u32,
```
Comment (Alexandcoats) [inline src/db/model/stardust.rs:30]: Can we make this a `bson::DateTime`?

File: bin/inx-chronicle/src/main.rs [hunk 7]
```diff
@@ -104,6 +104,7 @@ impl HandleEvent<Report<Broker>> for Launcher {
                                 cx.shutdown();
                             }
                         },
+                        other => log::warn!("Unhandled MongoDB error: {other}"),
```
Comment (Alexandcoats) [inline bin/inx-chronicle/src/main.rs:107]: Should call `cx.shutdown()` here, since we're not handling the error.

File: Cargo.toml (hunk 1 above)
Comment (grtlr) [reply Cargo.toml:3]: I will create tags on GitHub that will build the Docker image automatically (with the tags as semver)

File: src/db/model/stardust.rs (hunk 5 above)
Comment (grtlr) [reply src/db/model/stardust.rs:11]: All of these: https://github.com/iotaledger/inx/blob/develop/proto/inx.proto#L128-L153?

Comment (Alexandcoats) [reply src/db/model/stardust.rs:11]: Yeah, except maybe parents because that's on the message too.

File: Cargo.toml [hunk 8]
```diff
@@ -21,16 +21,25 @@ path = "bin/inx-chronicle/src/main.rs"
...
-clap = { version = "3.1", default-features = false, features = [ "derive", "std" ] }
+clap = { version = "3.1", default-features = false, features = ["derive", "std"] }
 dotenv = { version = "0.15", default-features = false }
 env_logger = { version = "0.9", default-features = false, features = ["termcolor", "atty", "humantime"] }
 futures = { version = "0.3", default-features = false }
-inx = { git = "https://github.com/iotaledger/inx", version = "0.2", default-features = false }
+inx = { path = "../inx/rust", version = "0.2", default-features = false, features = ["types"] }
 log = { version = "0.4", default-features = false }
-mongodb = { version = "2.1", default-features = false, features = ["tokio-runtime"] }
+mongodb = { version = "2.1", default-features = false, features = ["bson-chrono-0_4", "tokio-runtime"] }
```
Comment (Alexandcoats) [inline Cargo.toml:30]: The `bson-chrono-0_4` feature should not be needed here.

Comment (grtlr) [reply Cargo.toml:30]: You're right, thanks!

**Pull request closed without merging**